sdk/core/Azure.Core/src/Shared/OperationInternal.cs

```diff
@@ -68,32 +68,22 @@ internal class OperationInternal : OperationInternalBase
         /// </param>
         /// <param name="scopeAttributes">The attributes to use during diagnostic scope creation.</param>
         /// <param name="fallbackStrategy">The fallback delay strategy when Retry-After header is not present.  When it is present, the longer of the two delays will be used. Default is <see cref="ConstantDelayStrategy"/>.</param>
+        /// <param name="finalState">Final state of the operation for the case when underlying service request is completed before LRO instance is created.</param>
         public OperationInternal(
             ClientDiagnostics clientDiagnostics,
             IOperation operation,
             Response rawResponse,
             string? operationTypeName = null,
             IEnumerable<KeyValuePair<string, string>>? scopeAttributes = null,
-            DelayStrategy? fallbackStrategy = null)
-            :base(clientDiagnostics, rawResponse, operationTypeName ?? operation.GetType().Name, scopeAttributes, fallbackStrategy)
+            DelayStrategy? fallbackStrategy = null,
+            OperationState? finalState = null)
```

Comment:
Do we worry about the conflict between the response in this final state and the response as the parameter?  Should we take one or the other?

Reply:
@heaths , is it possible that these responses are different in keyvault?

Reply:
For the operations you changed, they are all pseudo-LROs anyway. The service backed endpoints aren't actually LROs, but we poll the final resource URI until it's not 404. Given the response and the final state should be agreed in that case, I don't think it should be possible.